app/main/views.py

```diff
@@ -120,10 +131,10 @@ def update(service_id, section):
         )
 
     if form.errors:
+        service.update(form.dirty_data)
```

Comment:
Right now pricing is the only "composite" field that we have. But this is the second time that the idea of having some kind of post-validation layer has come up. Maybe worth talking about on Tuesday?